Teradata/covalent
# fix(): products ng build/serve --prod build issue

## Description
The most recent version of covalent at the time of this writing [v1.0.0-rc.3](https://github.com/teradata/covalent/compare/v1.0.0-rc.2...v1.0.0-rc.3) converted Covalent distribution to Angular package format. Products that used this new dist was failing when running `--prod` flag on `ng serve` and `ng build` commands.

### What's included?

- Fixed `--prod` flag usage on products consuming Covalent (issue was invalid `covalent-core.metadata.json` generating from ngc usage inside ngpackagr which is used for aot when using the `--prod` flag).

#### Test Steps (Covalent build)
- Verify you are using node v8.9.x
- `npm run reinstall`
- To make the next steps a little easier go to `scripts/build-release` and add this code to the bottom of the file to `npm pack` instead of doing it manually:
```
cd deploy/platform/core
npm pack
cd ../dynamic-forms
npm pack
cd ../highlight
npm pack
cd ../http
npm pack
cd ../markdown
npm pack
```
- run `npm run build:release`
#### Test Steps (Product build)
- Clone Ecosystem Services UI repo into the directory above this one.
- `yarn install`
- `yarn remove @covalent/core`
- `yarn add <file-path-to-*.tgz>` (EX: `yarn add ../covalent/deploy/platform/core/covalent-core-1.0.0-rc.3.tgz`)
- `npm rebuild node-sass`
- `yarn upgrade @angular/material`
- Verify you are using node v8.9.x. 
- Verify `./node_modules/@angular/cli/bin/ng build --aot --prod --sourcemap=false --build-optimizer` successfully runs
- Verify `./node_modules/@angular/cli/bin/ng serve --aot --prod --sourcemap=false --build-optimizer --proxy-config proxy.conf.json` successfully runs and you can access the local site [http://localhost:4200/](http://localhost:4200/)
- Modify the import paths (primary/secondary entry point usage) in app.module and trying the prod build again it should run successfully.
- Try these similar instruction on the AppCenter repo and verify the --prod flag works in that product too

#### General Tests for Every PR

- [ ] `npm run build:release` still works.

closes https://github.com/Teradata/covalent/issues/1099
